Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+    
+    /// Operations that happen after device tracking has started.
+    fileprivate func finalizeStart() {
+        // Hide the camera image if necessary.
+        arSCNView.isHidden = !renderVideoFeed
+        
+        // Run the ARSession.
+        arSCNView.session.run(arConfiguration, options:.resetTracking)
+        didStartOrFailWithError(nil)
+    }
+
+    /// Adds subView to superView with appropriate constraints.
+    ///
+    /// - Parameter subview: The subView to add.
+    fileprivate func addSubviewWithConstraints(_ subview: UIView) {
+        // Add subview to view and setup constraints.
+        addSubview(subview)
+        subview.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            subview.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            subview.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            subview.topAnchor.constraint(equalTo: self.topAnchor),
+            subview.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+            ])
+    }
+    
+    /// Start the locationManager with undetermined access.
+    fileprivate func startWithAccessNotDetermined() {
+        if (Bundle.main.object(forInfoDictionaryKey: "NSLocationWhenInUseUsageDescription") != nil) {
+            locationManager.requestWhenInUseAuthorization()
+        }
+        if (Bundle.main.object(forInfoDictionaryKey: "NSLocationAlwaysUsageDescription") != nil) {
+            locationManager.requestAlwaysAuthorization()
+        }
+        else{
+            didStartOrFailWithError(ArcGISARView.clError.missingPListKey)
+        }
+    }
+    
+    /// Start updating the location and heading via the locationManager.
+    fileprivate func startUpdatingLocationAndHeading() {
+        locationManager.startUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.startUpdatingHeading()
+        }
+    }
+    
+    /// Stop updating location and heading.
+    fileprivate func stopUpdatingLocationAndHeading() {
+        locationManager.stopUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.stopUpdatingHeading()
+        }
+    }
+
+    /// Start the locationManager with denied access.
+    fileprivate func startWithAccessDenied() {
+        didStartOrFailWithError(ArcGISARView.clError.accessDenied)
+    }
+    
+    /// Start the locationManager with authorized access.
+    fileprivate func startWithAccessAuthorized() {
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Potential notification to the user of an error starting device tracking.
+    ///
+    /// - Parameter error: The error that occurred when starting tracking.
+    fileprivate func didStartOrFailWithError(_ error: Error?) {
+        if !notifiedStartOrFailure, let error = error {
+            // TODO: present error to user...
+            print("didStartOrFailWithError: \(String(reflecting:error))")
```

Comment:
do you want to leave print statements in?

Reply:
I deleted the `didStartOrFailWithError:` one out, but just commented out the `frameRate` one, as that will be useful during development.